Migrate BUSCO to nf-core module (#730)
nf-core/mag: docs/usage.md

When supplying pre-computed assemblies, reads **must** also be provided in the CSV input format to `--input`, and should be the reads used to build the assemblies, i.e., adapter-removed, run-merged etc.. Preprocessing steps will not be ran on raw reads when pre-computed assemblies are supplied. As long reads are only used for assembly, any long read fastq files listed in the reads CSV are ignored.

> ### Databases
> 
> #### BUSCO

Reviewer: ```suggestion
### Databases

nf-core/mag uses multiple tools that require additional databases.
The pipeline will download these databases for you if not supplied via a pipeline-level parameter.
However, we generally recommend you download these databases manually once, and place them in a long-term storage directory or cache from where you can re-use the databases across runs.
Here we provide specific additional information for some of the downloading databases which can be more tricky to prepare.

For any other database, or in doubt, please check the [parameters page](https://nf-co.re/mag/parameters).

#### BUSCO
```

> BUSCO can download lineage datasets automatically as it needs them, but this process can be slow if the internet connection is unstable, and may lead to redundant downloads across different samples. To avoid this, you can provide a local BUSCO lineage dataset using the --busco_db parameter. The local directory must follow a specific structure for BUSCO to recognize it.

Reviewer: ```suggestion
BUSCO can download lineage datasets automatically as it needs them, but this process can be slow if the internet connection is unstable, and may lead to redundant downloads across different samples.
To avoid this, you can provide a local copy of a BUSCO lineage dataset using the --busco_db parameter.
The local directory must follow a specific structure for BUSCO to recognize it.
```

To prepare the lineage dataset, the recommended method is to use BUSCO's built-in download functionality before running the pipeline:

```bash
busco --download_path <your_db> --download <lineage>
```

> This command downloads the specified lineage into the <your_db> directory and creates the required directory structure. <lineage> can be any supported dataset, such as `alphaproteobacteria_odb12`. You can also use `prokaryota` or `all` to download multiple lineages, which is necessary for automatic lineage selection in offline mode.

Reviewer: ```suggestion
This command downloads the specified lineage into the `<your_db>/` directory and creates the required directory structure.
`<lineage>` can be any [supported dataset](https://busco-data.ezlab.org/v5/data/lineages/), such as `alphaproteobacteria_odb12`.
You can also specify `prokaryota` or `all` to download multiple lineages, which is necessary for automatic lineage selection in offline mode.
```

> Alternatively, you can manually download a lineage tarball from [https://busco-data.ezlab.org/v5/data/lineages/](https://busco-data.ezlab.org/v5/data/lineages/), extract it, and place it in the appropriate location: `<your_db>/lineages/<taxa>_odb<XX>`. The lineage directory (e.g., bacteria_odb12) should contain files such as `datase.cfg` and a `hmms/` subdirectory at the top level. Then, you must provide `--busco_db <your_db>` to the pipeline.

Reviewer: ```suggestion
Alternatively, you can manually download a lineage tarball from [https://busco-data.ezlab.org/v5/data/lineages/](https://busco-data.ezlab.org/v5/data/lineages/), extract it, and place it in the appropriate location: `<your_db>/lineages/<taxa>_odb<XX>`.
The lineage directory (e.g., bacteria_odb12) should contain files such as `datase.cfg` and a `hmms/` subdirectory at the top level.
Then, you must provide `--busco_db <your_db>/` to the pipeline.
```

> 

Reviewer: ```suggestion

### CAT

> [! WARNING]
> This database is very large at ~180 GB!
> This can take a long time, so we strongly recommend downloading and unzipping prior the pipeline run.

This database can be downloaded from the [CAT developers' website](https://tbb.bio.uu.nl/bastiaan/CAT_prepare/), which is based in the Netherlands (and could be slow for other regions of the world)

### GTDB

> [! WARNING]
> This database is very large at ~110 GB!
> This can take a long time, so we strongly recommend downloading and unzipping prior the pipeline run.

This database can be downloaded from the [GTDB developers' website](default: https://data.gtdb.ecogenomic.org/releases/release220/220.0/auxillary_files/gtdbtk_package/full_package/), which is based in Australia (and could be slow for other regions of the world).
The developers also offer a 'split' archive of 10GB files that can be downloaded more stably and subsequently can be manually combined after [here](https://data.gtdb.ecogenomic.org/releases/release220/220.0/auxillary_files/gtdbtk_package/split_package/).
More documentation can be seen [here](https://ecogenomics.github.io/GTDBTk/installing/index.html#gtdb-tk-reference-data).

```

Reviewer 2: Get off your phone Jim, you’re on holiday, etc…

Should that read “database.cfg”, not “datase.cfg”?

```bash
busco --download_path <your_db> --download <lineage>
```

This command downloads the specified lineage into the `<your_db>/` directory and creates the required directory structure.
`<lineage>` can be any [supported dataset](https://busco-data.ezlab.org/v5/data/lineages/), such as `alphaproteobacteria_odb12`.
You can also specify `prokaryota` or `all` to download multiple lineages, which is necessary for automatic lineage selection in offline mode.

Alternatively, you can manually download a lineage tarball from [https://busco-data.ezlab.org/v5/data/lineages/](https://busco-data.ezlab.org/v5/data/lineages/), extract it, and place it in the appropriate location: `<your_db>/lineages/<taxa>_odb<XX>`.
The lineage directory (e.g., bacteria_odb12) should contain files such as `datase.cfg` and a `hmms/` subdirectory at the top level.
Then, you must provide `--busco_db <your_db>/` to the pipeline.

### CAT

> [! WARNING]
> This database is very large at ~180 GB!
> This can take a long time, so we strongly recommend downloading and unzipping prior the pipeline run.

> This database can be downloaded from the [CAT developers' website](https://tbb.bio.uu.nl/bastiaan/CAT_prepare/), which is based in the Netherlands (and could be slow for other regions of the world)

Reviewer: ```suggestion
This database can be downloaded from the [CAT developers' website](https://tbb.bio.uu.nl/bastiaan/CAT_prepare/), which is based in the Netherlands (and could be slow for other regions of the world).
```

### GTDB

> [! WARNING]
> This database is very large at ~110 GB!
> This can take a long time, so we strongly recommend downloading and unzipping prior the pipeline run.

This database can be downloaded from the [GTDB developers' website](default: https://data.gtdb.ecogenomic.org/releases/release220/220.0/auxillary_files/gtdbtk_package/full_package/), which is based in Australia (and could be slow for other regions of the world).

> The developers also offer a 'split' archive of 10GB files that can be downloaded more stably and subsequently can be manually combined after [here](https://data.gtdb.ecogenomic.org/releases/release220/220.0/auxillary_files/gtdbtk_package/split_package/).

Reviewer: ```suggestion
The developers also offer a 'split' archive of 10GB files that can be downloaded more stably from  [here](https://data.gtdb.ecogenomic.org/releases/release220/220.0/auxillary_files/gtdbtk_package/split_package/) and subsequently (manually) combined after.
```

For any other database, or in doubt, please check the [parameters page](https://nf-co.re/mag/parameters).

#### BUSCO

BUSCO can download lineage datasets automatically as it needs them, but this process can be slow if the internet connection is unstable, and may lead to redundant downloads across different samples.
To avoid this, you can provide a local copy of a BUSCO lineage dataset using the `--busco_db` parameter.
The local directory must follow a specific structure for BUSCO to recognize it.

To prepare the lineage dataset, the recommended method is to use BUSCO's built-in download functionality before running the pipeline:

```bash
busco --download_path <your_db> --download <lineage>
```

This command downloads the specified lineage into the `<your_db>/` directory and creates the required directory structure.
`<lineage>` can be any [supported dataset](https://busco-data.ezlab.org/v5/data/lineages/), such as `alphaproteobacteria_odb12`.
You can also specify `prokaryota` or `all` to download multiple lineages, which is necessary for automatic lineage selection in offline mode.

> Alternatively, you can manually download a lineage tarball from [https://busco-data.ezlab.org/v5/data/lineages/](https://busco-data.ezlab.org/v5/data/lineages/), extract it, and place it in the appropriate location: `<your_db>/lineages/<taxa>_odb<XX>`.

Reviewer: **suggestion:** 
```suggestion
Alternatively, you can manually download a specific lineage tarball from [https://busco-data.ezlab.org/v5/data/lineages/](https://busco-data.ezlab.org/v5/data/lineages/), extract it, and place it in the appropriate location: `<your_db>/lineages/<taxa>_odb<XX>`.
```

MetaBAT2 is run by default with a fixed seed within this pipeline, thus producing reproducible results.

By default, BUSCO runs in offline mode only when both a local database path is provided via `--busco_db` and a specific lineage is set with `--busco_db_lineage`.
Using auto lineage mode with BUSCO may lead to non-reproducible results, since the databases are frequently updated and automatic lineage selection depends on the version of the database used when running BUSCO.

> If reproducibility is crucial, you can enable the parameter `--busco_force_offline`.

Reviewer: ```suggestion
If you have supplied all available lineages within your `--busco_db`, but want BUSCO to auto-select the lineage, and are running in an offline context, you can enable the parameter `--busco_force_offline`, for maximum reproducibility.
```

MetaBAT2 is run by default with a fixed seed within this pipeline, thus producing reproducible results.

By default, BUSCO runs in offline mode only when both a local database path is provided via `--busco_db` and a specific lineage is set with `--busco_db_lineage`.

> Using auto lineage mode with BUSCO may lead to non-reproducible results, since the databases are frequently updated and automatic lineage selection depends on the version of the database used when running BUSCO.

Reviewer: **suggestion:** 
```suggestion
Using auto lineage mode with BUSCO may lead to non-reproducible results, since the databases are frequently updated and automatic lineage selection depends on the version of the database used when running BUSCO.
Therefore we strongly recommend downloading the lineages in advance and specifying the lineage to check against.
You can also download `all` lineages (see [Databases](#databases), and specify this in `--busco_db_lineages` to check against all lineages, if you're not sure which lineage to select.
```

I think?

> If you have supplied all available lineages within your `--busco_db`, but want BUSCO to auto-select the lineage, and are running in an offline context, you can enable the parameter `--busco_force_offline`, for maximum reproducibility.

Reviewer 2: ^

> If you have supplied all available lineages within your `--busco_db`, but want BUSCO to auto-select the lineage, and are running in an offline context, you can enable the parameter `--busco_force_offline`, for maximum reproducibility.

Reviewer: **suggestion:** 
```suggestion
```

MetaBAT2 is run by default with a fixed seed within this pipeline, thus producing reproducible results.

By default, BUSCO runs in offline mode only when both a local database path is provided via `--busco_db` and a specific lineage is set with `--busco_db_lineage`.
Using auto lineage mode with BUSCO may lead to non-reproducible results, since the databases are frequently updated and automatic lineage selection depends on the version of the database used when running BUSCO.

> If you have supplied all available lineages within your `--busco_db`, but want BUSCO to auto-select the lineage, and are running in an offline context, you can enable the parameter `--busco_force_offline`, for maximum reproducibility.

Reviewer 2: Is this still relevant?
